scripts/dev/upstream_merge/README.md

```diff
@@ -0,0 +1,144 @@
+# HOW TO USE THE MERGE SCRIPT
+
+This script automates the process of merging upstream changes, building images, and testing them on a virtual machine (VM). Follow the steps below to use the script effectively.
+
+---
+
+## **Steps to Run the Script**
+
+### **1. Navigate to the `nilrt` Directory**
+```bash
+cd ~/nilrt
+```
+
+### **2. Run the Script**
+
+#### **To Perform a Merge**
+This will merge upstream changes, build the images, and test them:
+```bash
+python3 scripts/dev/upstream_merge/upstream_merge.py -c scripts/dev/upstream_merge/automation_conf.json
+```
+
+#### **To Skip the Merge and Only Build and Test**
+If you want to skip the merge step and directly proceed with building and testing:
+```bash
+python3 scripts/dev/upstream_merge/upstream_merge.py -c scripts/dev/upstream_merge/automation_conf.json -skip-merge True
+```
+
+---
+
+## **Configuration File: `automation_conf.json`**
+
+The script relies on a configuration file (`automation_conf.json`) to define various parameters. Below is an explanation of the fields in the configuration file:
+
+- **`conf_file_path`**:  
+  Path to the configuration file (default: `repos.conf`). This file must be located in the `/scripts/dev/upstream_merge/` folder.
+
+- **`force_checkout`**:  
+  Enables forceful checkout to the base branch if set to `True`.
+
+- **`forks`**:  
+  A dictionary where the keys are local repository names (as mentioned in `repos.conf`) and the values are the URLs of your forks.
```

Comment:
Will you create dedicated forks for automating stuff? And mention here that the fork URLs are set to these by default.